### PR TITLE
Object highlighter and protips list screen fix

### PR DESCRIPTION
**Pull request by MaxIsJoe**
I was working on a super-secret-update for this new years and I needed the object highlighter that was planned for the protips so here is it and some fixes as well

### Changelog

CL: [New] Added a new system to highlight any game object on the client's side which is utilized by the protips only for now.
CL: [Fix] The `triggerOnce` condition for protips did not correctly save under some circumstances, now this has been fixed.
CL: [Fix] The protips list screen has had its viewport fixed.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Managers/GlobalHighlighterManager.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+			Instance.StartCoroutine(Instance.Search(searchName.ToLower()));
+		}
+
+		private IEnumerator Search(string searchName)
+		{
+			var index = -1;
+			foreach (var possibleTarget in FindObjectsOfType<GameObject>())
+			{
+				index++;
+				if (maximumSearchIndexesPerFrame > index)
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Managers/GlobalHighlighterManager.cs]: isn't the logic inverted here?

File: UnityProject/Assets/Scripts/Items/Attributes.cs [hunk 2]
```diff
@@ -274,4 +281,23 @@ public void ServerSetArticleDescription(string desc)
...
+			names.AddRange(c.GetTraits().Select(trait => trait.name));
+		}
+		return names;
+	}
+
+	public void HighlightObject()
+	{
+		if (PlayerManager.LocalPlayerObject == null || gameObject.IsAtHiddenPos()) return;
+		if (Vector2.Distance(gameObject.AssumedWorldPosServer(), PlayerManager.LocalPlayerObject.AssumedWorldPosServer()) > MINIMUM_HIGHLIGHT_DISTANCE) return;
+		Instantiate(GlobalHighlighterManager.HighlightObject, this.transform, false);
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Items/Attributes.cs:301]: when does the highlight go away? I see no code to unhighlight the object.

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Attributes.cs:301]: It unhighlights it self from the other script

File: UnityProject/Assets/Scripts/Managers/GlobalHighlighterManager.cs (hunk 1 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Managers/GlobalHighlighterManager.cs]: fixed

File: UnityProject/Assets/Scripts/Items/Attributes.cs (hunk 2 above)
Comment (corp-0) [reply UnityProject/Assets/Scripts/Items/Attributes.cs:301]: what other script?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Attributes.cs:301]: https://github.com/unitystation/unitystation/pull/9439/files#diff-34627acdc683074eeaf307bdc684c16bff305c4892d17c0504d7d37e2dfb4c7b

**Comment by corp-0** [comment]
> GameManager shouldn't have QuickLoad enabled!at Tests.Config.GameManagerTest.CheckQuickLoad () [0x000a3] in /github/workspace/UnityProject/Assets/Tests/Config/GameManagerTest.cs:37

You pushed quickload enabled.

**Comment by MaxIsJoe** [comment]
> You pushed quickload enabled.

hewwo, thwis has been updated awhile ago can you pwease merge uwu

**Review by corp-0 — APPROVED** [review]

**Pull request merged**